src/containers/SortingsTable.tsx

```diff
@@ -1,72 +1,60 @@
-import React, { Dispatch, FunctionComponent, useContext, useEffect, useMemo, useRef } from 'react';
+import React, { Dispatch, FunctionComponent, useCallback, useMemo } from 'react';
 import { connect, MapDispatchToProps, MapStateToProps } from 'react-redux';
-import { Link } from 'react-router-dom';
-import { deleteSortings, setSortingInfo } from '../actions';
+import { deleteSortings } from '../actions';
+import { WorkspaceInfo } from '../AppContainer';
 import NiceTable from '../components/NiceTable';
-import { createCalculationPool, HitherContext } from '../extensions/common/hither';
-import { getPathQuery } from '../kachery';
+import { useSortingInfos } from '../extensions/common/getRecordingInfo';
 import { RootAction, RootState } from '../reducers';
-import { Sorting, SortingInfo } from '../reducers/sortings';
-import { WorkspaceInfo } from '../reducers/workspaceInfo';
+import { Sorting } from '../reducers/sortings';
+import { WorkspaceRouteDispatch } from './WorkspaceView';
 
 
 
 interface StateProps {
-    workspaceInfo: WorkspaceInfo
 }
 
 interface DispatchProps {
     onDeleteSortings: (sortingIds: string[]) => void
-    onSetSortingInfo: (a: { sortingId: string, sortingInfo: SortingInfo }) => void
 }
 
 interface OwnProps {
     sortings: Sorting[]
+    workspaceRouteDispatch: WorkspaceRouteDispatch
+    workspaceInfo: WorkspaceInfo
 }
 
 type Props = StateProps & DispatchProps & OwnProps
 
-const calculationPool = createCalculationPool({maxSimultaneous: 6})
-
-const SortingsTable: FunctionComponent<Props> = ({ sortings, onDeleteSortings, onSetSortingInfo, workspaceInfo }) => {
-    const hither = useContext(HitherContext)
-    const { workspaceName, feedUri, readOnly } = workspaceInfo;
-    const infosInProgress = useRef(new Set<string>())
-
-    useEffect(() => {
-        for (const sor of sortings) {
-            if ((!sor.sortingInfo) && (!infosInProgress.current.has(sor.sortingId))) {
-                infosInProgress.current.add(sor.sortingId)
-                hither.createHitherJob(
-                    'createjob_get_sorting_info',
-                    { sorting_object: sor.sortingObject, recording_object: sor.recordingObject },
-                    {
-                        useClientCache: true,
-                        calculationPool: calculationPool
-                    }
-                ).wait().then(sortingInfo => {
-                    onSetSortingInfo({ sortingId: sor.sortingId, sortingInfo: sortingInfo });
-                }).catch((err: Error) => {
-                    console.error(err);
-                    return;
-                })
-            }
-        }
-    }, [sortings, hither, onSetSortingInfo])
+const SortingsTable: FunctionComponent<Props> = ({ sortings, onDeleteSortings, workspaceInfo, workspaceRouteDispatch }) => {
+    const { readOnly } = workspaceInfo;
+
+    const handleViewSorting = useCallback((sorting: Sorting) => {
+        workspaceRouteDispatch({
```

Comment:
This handler is called when the user clicks on a sorting... it takes us to the SortingView page which is essentially the MVSortingView. This routing is handled via `workspaceRouteDispatch` which will be described further below.